sdks/java/io/kafka/src/test/java/org/apache/beam/sdk/io/kafka/KafkaIOIT.java

```diff
@@ -166,6 +174,41 @@ public void testKafkaIOReadsAndWritesCorrectlyInStreaming() throws IOException {
     }
   }
 
+  @Test
+  public void testKafkaIOReadsAndWritesCorrectlyInBatchNullKey() throws IOException {
+    List<String> values = new ArrayList<>();
+    for (int i = 0; i < 100; i++) {
+      values.add("value" + Integer.toString(i));
+    }
+    PCollection<String> writeInput =
+        writePipeline.apply(Create.of(values)).setCoder(StringUtf8Coder.of());
+
+    writeInput.apply(
+        KafkaIO.<byte[], String>write()
+            .withBootstrapServers(options.getKafkaBootstrapServerAddresses())
+            .withTopic(options.getKafkaTopic())
+            .withValueSerializer(StringSerializer.class)
+            .values());
+
+    PCollection<String> readOutput =
+        readPipeline
+            .apply("Read from bounded Kafka", readFromKafkaNullKey())
+            .apply("Materialize input", Reshuffle.viaRandomKey())
+            .apply(
+                "Map records to strings", MapElements.via(new MapKafkaRecordsToStringsNullKey()));
```

Comment:
You can use a lambda here, like: https://github.com/apache/beam/blob/243128a8fc52798e1b58b0cf1a271d95ee7aa241/sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/meta/provider/kafka/PayloadSerializerKafkaTable.java#L53-L54